pkg/controller/unifiedpushserver/unifiedpushserver.go

```diff
@@ -0,0 +1,211 @@
+package unifiedpushserver
+
+import (
+	"fmt"
+
+	aerogearv1alpha1 "github.com/aerogear/unifiedpush-operator/pkg/apis/aerogear/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func newOauthProxyServiceAccount(cr *aerogearv1alpha1.UnifiedPushServer) *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name,
+			Namespace: cr.Namespace,
+			Annotations: map[string]string{
+				"serviceaccounts.openshift.io/oauth-redirectreference.ups": fmt.Sprintf("{\"kind\":\"OAuthRedirectReference\",\"apiVersion\":\"v1\",\"reference\":{\"kind\":\"Route\",\"name\":\"%s-unifiedpush-proxy\"}}", cr.Name),
+			},
+		},
+	}
+}
+
+func newOauthProxySaRoleBinding(cr *aerogearv1alpha1.UnifiedPushServer) *rbacv1.RoleBinding {
+	return &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name,
+			Namespace: cr.Namespace,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind: "ServiceAccount",
+				Name: cr.Name,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			Kind: "Role",
+			Name: "admin",
+		},
+	}
+}
+
+func newOauthProxyService(cr *aerogearv1alpha1.UnifiedPushServer) (*corev1.Service, error) {
+	return &corev1.Service{
+		ObjectMeta: objectMeta(cr, "unifiedpush-proxy"),
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"app":     cr.Name,
+				"service": "ups",
+			},
+			Ports: []corev1.ServicePort{
+				corev1.ServicePort{
+					Name:     "web",
+					Protocol: corev1.ProtocolTCP,
+					Port:     80,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 4180,
+					},
+				},
+			},
+		},
+	}, nil
+}
+
+func newUnifiedPushServerDeployment(cr *aerogearv1alpha1.UnifiedPushServer) *appsv1.Deployment {
+	labels := map[string]string{
```

Comment:
It could be in a helper for the CR since will be used for all impl which will create/manage the resources.

Reply:
Thanks for the feedback!

I've already got a function here in the util file: https://github.com/aerogear/unifiedpush-operator/pull/6/files#diff-5e52281b0323c385631f2ec4a8e2be64R33

In this case I wanted a slightly different service value so I just created this local variable instead.

Reply:
Just to clarifies, the label is specific by CR and the units.go not then because of this in MSS Oper each CR has your own helper.